nhs/data/geography.py

```diff
@@ -0,0 +1,295 @@
+    """Parallel process point data."""
+    chunks = np.array_split(pnts_gdf, num_cores)
+    with Pool(num_cores) as pool:
+        all_results = pool.map(lambda chunk: process_chunk(map_data, chunk), chunks)
+    return pl.concat(all_results)
+
+
+def shuffle_coordinates(
+    coordinates: List[Tuple[float, float]], seed: int
+) -> List[Tuple[float, float]]:
+    """
+    Shuffle a list of (x, y) coordinate pairs.
+
+    Args:
+        coordinates (List[Tuple[float, float]]): List of coordinate pairs to shuffle.
+        seed (int): Seed for the random number generator to ensure reproducibility.
+
+    Returns:
+        List[Tuple[float, float]]: Shuffled list of coordinate pairs.
+    """
+    return (
+        pl.DataFrame(
+            {"x": [c[0] for c in coordinates], "y": [c[1] for c in coordinates]}
+        )
+        .sample(frac=1, seed=seed)
+        .to_numpy()
+        .tolist()
+    )
+
+
+def assign_shuffled_coordinates(
+    df: pl.DataFrame, area_code: str, shuffled_coords: List[Tuple[float, float]]
+) -> pl.DataFrame:
+    """
+    Assign shuffled coordinates to the dataframe for a specific area_code.
+
+    Args:
+        df (pl.DataFrame): Input dataframe.
+        area_code (str): The area code to which shuffled coordinates should be assigned.
+        shuffled_coords (List[Tuple[float, float]]): List of shuffled coordinate pairs.
+
+    Returns:
+        pl.DataFrame: Dataframe with shuffled coordinates assigned for the specified area_code.
+    """  # noqa: E501
+    return df.with_columns(
+        [
+            pl.when(pl.col("area_code") == area_code)
+            .then(pl.Series([coord[0] for coord in shuffled_coords]))
+            .otherwise(pl.col("x"))
+            .alias("x"),
+            pl.when(pl.col("area_code") == area_code)
+            .then(pl.Series([coord[1] for coord in shuffled_coords]))
+            .otherwise(pl.col("y"))
+            .alias("y"),
+        ]
+    )
+
+
+def random_distribution(
+    df: pl.DataFrame, num_iterations: int = 10, seed: int = 42
+) -> pl.DataFrame:
+    """
+    Randomly shuffle lat-long pairs within their respective areas, maintaining uniqueness.
+
+    This function shuffles coordinates within each area_code, ensuring that the number of unique
+    coordinates remains the same before and after shuffling. It uses a seed for reproducibility.
+
+    This was used in testing for Schelling's model of segregation, which worked with very limited
+    success (something like 30minutes to move 5 people).
+
+    Remains here in case of future expansion into distributions.
+
+    Args:
+        df (pl.DataFrame): Input dataframe containing 'x', 'y', and 'area_code' columns.
+        num_iterations (int, optional): Number of shuffling iterations. Defaults to 10.
+        seed (int, optional): Seed for the random number generator. Defaults to 42.
+
+    Returns:
+        pl.DataFrame: Dataframe with shuffled coordinates, maintaining original uniqueness within each area_code.
```

Comment:
Not pandas stype docstring. btw what does "maintaining original uniqueness" mean? I assume shuffling by default won't introduce duplicate data?